matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/pushers/AddPusherWorker.kt

```diff
@@ -58,50 +47,17 @@ internal class AddPusherWorker(context: Context, params: WorkerParameters)
             return Result.failure()
         }
         return try {
-            setPusher(pusher)
+            addPusherTask.execute(AddPusherTask.Params(pusher))
             Result.success()
         } catch (exception: Throwable) {
             when (exception) {
                 is Failure.NetworkConnection -> Result.retry()
-                else                         -> {
```

Comment:
due to extracting the logic to a task there's now a slight change in behaviour, the `AddPusherTask` will always update any existing pusher state on failure instead of `exception != NetworkConnection` as the task can be used in non worker contexts (such as the settings page)